### PR TITLE
Versions

### PR DESCRIPTION
**Pull request by cydparser**

### CODE REVIEW

File: src/Codex.hs [hunk 1]
```diff
@@ -48,9 +48,10 @@ import Codex.Project
 replace :: String -> String -> String -> String
 replace a b c = Text.unpack $ Text.replace (Text.pack a) (Text.pack b) (Text.pack c)
 
-md5hash :: String -> String
-md5hash = concatMap (Printf.printf "%02x") . C8.unpack . MD5.hash . C8.pack
-
+sha1hash :: String -> String
+sha1hash = Text.unpack . Text.decodeUtf8 . sha1 . Text.encodeUtf8 . Text.pack
+  where
+    sha1 = BA.convertToBase BA.Base64 . Crypto.hashWith Crypto.SHA1
```
Comment (parsonsmatt) [inline src/Codex.hs]: Why are we changing this? It's going to require rehashing everything for every user of the code. Is there a problem that this solves?

File: Makefile [hunk 2]
```diff
@@ -1,7 +0,0 @@
-build: build-7.10
-
-build-7.10:
-	STACK_YAML="stack-7.10.yaml" stack build
-
-build-8.0:
-	STACK_YAML="stack-8.0.yaml" stack build
```
Comment (parsonsmatt) [inline Makefile]: Why is this deleted? It should be updated or left alone.

File: .travis.yml [hunk 3]
```diff
@@ -55,10 +55,6 @@ matrix:
     compiler: ": #stack default"
     addons: {apt: {sources: [hvr-ghc],packages: [libgmp-dev,dnsutils,cabal-install-2.2]}}
 
-  - env: BUILD=stack ARGS="--resolver lts-6"
-    compiler: ": #stack 7.10.3"
-    addons: {apt: {sources: [hvr-ghc],packages: [libgmp-dev,dnsutils,cabal-install-1.22]}}
```
Comment (parsonsmatt) [inline .travis.yml]: Why are we dropping support?

File: stack-7.10.yaml [hunk 4]
```diff
@@ -1,15 +0,0 @@
-resolver: lts-6.16
```
[the remark this answers is not in the document]
Comment (cydparser) [reply stack-7.10.yaml]: I removed the stack.yaml files because they appeared to be unmaintained (e.g. lts-6.16 instead of lts-6.35), some contained unnecessary `extra-deps`, and they are not used by CI. One can use `--resolver` instead of `--stack-yaml` with the added benefit of running the same command that CI runs.

File: src/Codex.hs (hunk 1 above)
Comment (cydparser) [reply src/Codex.hs]: The change was motivated by removing `cryptohash` (deprecated). The decision to switch to sha1 as well was made in a moment of over caffeination: I'll put md5 back.

File: Makefile (hunk 2 above)
Comment (cydparser) [reply Makefile]: Why does it exist? It looks like cruft.

File: .travis.yml (hunk 3 above)
Comment (cydparser) [reply .travis.yml]: Dropping this resolver allowed me to bump the lower bounds of `directory` and remove a warning. How many versions of GHC is it customary to support for an application?

File: Makefile (hunk 2 above)
Comment (bitemyapp) [reply Makefile]: @cydparser it's not cruft, it spares the user typing and makes them aware of the option to build that way. There is no way it saves people time net-net removing this versus leaving it in unless you're actually deleting and purging the Stack GHC 8.0 build.

File: .travis.yml (hunk 3 above)
Comment (bitemyapp) [reply .travis.yml]: @cydparser >How many versions of GHC is it customary to support for an application?

3 for libraries but tools often support more.

File: stack-7.10.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (parsonsmatt) [reply stack-7.10.yaml]: `--resolver` doesn't let you specify extra-deps, though

File: src/Codex.hs (hunk 1 above)
Comment (parsonsmatt) [reply src/Codex.hs]: I'd be fine with SHA1, but we shouldn't break old users of the tool. I think we could try SHA1 first, and if it fails, then we can do MD5. If MD5 succeeds, then we carry on, and eventually write a SHA1 hash to the file instead of MD5.

File: .travis.yml (hunk 3 above)
Comment (parsonsmatt) [reply .travis.yml]: > Dropping this resolver allowed me to bump the lower bounds of directory and remove a warning. 

Ok, but why are we bumping the lower bounds of `directory`? I'm not opposed to dropping support, but I'd like to ensure that it's worth it.

File: src/Codex.hs (hunk 1 above)
Comment (cydparser) [reply src/Codex.hs]: That is a good idea, but I'll leave it for a separate pull request.

**Comment by cydparser** [comment]
@parsonsmatt, any objections to this being merged now?
